src/link/data-access-link/useGetLinks.ts

```diff
@@ -0,0 +1,40 @@
+import { useCallback, useEffect } from "react";
+import { axiosInstance } from "../../sharing/util";
+import { mapLinksData } from "../../link/util-map/mapLinksData";
+import { useAsync } from "../../sharing/util";
+import { ALL_LINKS_ID } from "./constant";
+
+interface MapDataFormatProps {
+  id: string | number;
+  created_at: Date;
+  url: string;
+  image_source: string;
+  title: string;
+  description: string;
+}
+export const useGetLinks = (folderId = ALL_LINKS_ID) => {
+  const queryString = folderId === ALL_LINKS_ID ? "" : `?folderId=${folderId}`;
+  const getLinks = useCallback(
+    () => axiosInstance.get(`users/1/links${queryString}`),
+    [queryString]
+  );
+  const { execute, loading, error, data } = useAsync(getLinks);
+
+  useEffect(() => {
+    execute();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [folderId]);
+
+  const mapDataFormat = ({ id, created_at, url, image_source, title, description }: MapDataFormatProps) => ({
+    id,
+    createdAt: created_at,
+    imageSource: image_source,
+    url,
+    title,
+    description,
+  });
```

Comment:
💬  how about

```suggestion
  const mapDataFormat = ({ created_at, image_source, ...rest }: MapDataFormatProps) => ({
    createdAt: created_at,
    imageSource: image_source,
    ...rest
  });
```

이렇게 하면 api에 field가 추가되더라도 type만 변형 시켜주면 될 것 같습니다